### PR TITLE
chore: add e2e for GKE

### PR DESCRIPTION
**Pull request by aldor007**

### CODE REVIEW

File: e2e/tests/gke_cluster_zonal/main.tf [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+module "castai-gke-iam" {
+  source = "castai/gke-iam/castai"
+
+  project_id       = var.project_id
+  gke_cluster_name = var.cluster_name
+
+}
+
+module "castai-gke-cluster" {
+  source = "../../../../terraform-castai-gke-cluster"
```
Comment (matas-cast) [inline e2e/tests/gke_cluster_zonal/main.tf:30]: probably can use registry module?

File: e2e/gke_test.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+		res, err := castAIClient.ExternalClusterAPIGetClusterWithResponse(ctx, clusterID)
+		r.NoError(err)
+		if res != nil && res.JSON200 != nil && res.JSON200.Status != nil && *res.JSON200.Status == "ready" {
+			return true
+		}
+		return false
+	}, time.Minute*5, time.Second*30, "cluster doesn't become ready after timeout")
+
+	// Try to add node
+	// FIXME: not working yet
```
Comment (matas-cast) [inline e2e/gke_test.go:53]: What is the issue here?

File: e2e/tests/gke_cluster_zonal/main.tf (hunk 1 above)
Comment (aldor007) [reply e2e/tests/gke_cluster_zonal/main.tf]: Updated

**Comment by matas-cast** [comment]
Looks good to me (also seen the demo) but let's reach decision where to place it and run (Github CI on provider PR merge or our private repo)

**Pull request closed without merging**